src/rest/resources.ts

```diff
@@ -15,7 +15,13 @@ import * as Qs from 'qs';
 export interface FactoryResolver {
     v: string;
     source: string;
-    devfile: che.workspace.devfile.Devfile
+    devfile: che.workspace.devfile.Devfile;
+    scm_info: {
+        clone_url: string;
+        scm_provider: string;
+        branch: string;
```

Comment:
Is it require field?
```
   branch: string;
```
->
```
   branch?: string;
```

Reply:
let me check !

Reply:
so it seems the branch name is always there
https://github.com/mshaposhnik/che-server/blob/c5f529759a2d610c94f35c287a65df897929102f/core/che-core-api-model/src/main/java/org/eclipse/che/api/core/model/factory/ScmInfo.java

Probably because we want to know if it's for example 'main' or 'master' or something else

Reply:
fixed